app/app.js

```diff
@@ -137,30 +139,44 @@ window.run = function() {
             });
         };
 
-        var appLayoutPromise = Application.getOSInformation().then(function(osInfo) {
-            if (BaseConfig.useTabLayout) {
-                return createTabBarLayout(counterBadgeControllerPromise);
-            } else {
-                return createDrawerLayout(counterBadgeControllerPromise);
-            }
-        });
+        var createLayout = function() {
+            return BaseConfig.useTabLayout
+                ? createTabBarLayout()
+                : createDrawerLayout();
+        };
 
-        // Show welcome modal only after layout is created for proper
-        // bookkeeping of the active view.
-        Promise.join(
-            appLayoutPromise,
-            welcomeModalControllerPromise,
-        function(menuController, welcomeModalController) {
-            // The welcome modal can be configured to show only
-            // once -- on initial startup, by passing in the
-            // parameter `{forced: false}` below
-            welcomeModalController.show({forced: true});
-
-            // Deep linking services will enable deep linking on startup
-            // and while running
-            // It will open the deep link in the current active tab
-            deepLinkingServices = new DeepLinkingServices(menuController);
-        });
+        var initMainLayout = function() {
+            return createLayout().then(function(menuController) {
+                // Deep linking services will enable deep linking on startup
+                // and while running it will open the deep link in the current
+                // active tab
+                deepLinkingServices = new DeepLinkingServices(menuController);
+            });
+        };
+
+        var runApp = function() {
```

Comment:
When preview resolves, it passes the URL that was previewed. Can we pass that through so that the rest of the app can base other "app urls" off of this?